elements/howto-tabs/howto-tabs.js

```diff
@@ -284,24 +311,81 @@
       // changes its role to `tab`.
       this.setAttribute('role', 'tab');
       if (!this.id)
-        this.id = `howto-tabs-tab-generated-${dashTabCounter++}`;
+        this.id = `howto-tabs-tab-generated-${howtoTabCounter++}`;
+
+      // Set a well-defined initial state.
+      this.selected = false;
+      this.setAttribute('aria-selected', 'false');
+      this.setAttribute('tabindex', -1);
+    }
+
+    /**
+     * Properties and their corresponding attributes should mirror one another.
+     * To this effect, the property setter for `selected` handles truthy/falsy
+     * values and reflects those to the state of the attribute. It’s important
+     * to note that there are no side effects taking place in the property
+     * setter. For example, the setter does not set `aria-selected`. Instead,
+     * that work happens in the `attributeChangedCallback`. As a general rule,
+     * make property setters very dumb, and if setting a property or attribute
+     * should cause a side effect (like setting a corresponding ARIA attribute)
+     * do that work in the `attributeChangedCallback`. This will avoid having to
+     * manage complex attribute/property reentrancy scenarios.
+     */
+    attributeChangedCallback() {
+      const value = this.hasAttribute('selected');
+      this.setAttribute('aria-selected', value);
+      this.setAttribute('tabindex', value?0:-1);
+    }
+
+    set selected(value) {
+      value = Boolean(value);
+      if(value)
+        this.setAttribute('selected', '');
+      else
+        this.removeAttribute('selected');
+    }
+
+    get selected() {
+      return this.hasAttribute('selected');
     }
   }
   window.customElements.define('howto-tabs-tab', HowtoTabsTab);
 
-  let dashPanelCounter = 0;
+  let howtoPanelCounter = 0;
   /**
    * `HowtoTabsPanel` is a panel for a `<howto-tabs>` tab panel.
    */
   class HowtoTabsPanel extends HTMLElement {
+    static get observedAttributes() {
+      return ['hidden'];
```

Comment:
I added a test helper `isHidden` because setting `hidden` does not set `aria-hidden` (in terms of the attribute), and my e2e tests were relying solely on `aria-hidden` up until this point.